test/controllers/util/CustomMetadataUtilsSpec.scala

```diff
@@ -0,0 +1,143 @@
+        namesOfPropertiesToGet.contains(propertyRetrieved.name)
+    )
+    propertiesRetrievedEqualPropertiesRequested should equal(true)
+  }
+
+  "getCustomMetadataProperties" should "throw an 'NoSuchElementException' if any properties requested are not present" in {
+    val namesOfPropertiesToGet = Set("TestProperty11", "TestProperty3")
+
+    val thrownException: NoSuchElementException =
+      the[NoSuchElementException] thrownBy customMetadataUtils.getCustomMetadataProperties(namesOfPropertiesToGet)
+
+    thrownException.getMessage should equal("key not found: TestProperty11")
+  }
+
+  "getValuesOfProperties" should "return the values for a given property" in {
+    val namesOfPropertiesAndTheirExpectedValues = Map(
+      "TestProperty5" -> List("TestValue 1"),
+      "TestProperty5" -> List("TestValue 1", "TestValue 2", "TestValue 3"),
+      "TestProperty5" -> List("TestValue 1")
+    )
+
+    val actualPropertiesAndTheirValues: Map[String, List[CustomMetadata.Values]] =
+      customMetadataUtils.getValuesOfProperties(namesOfPropertiesAndTheirExpectedValues.keys.toSet)
+
+    namesOfPropertiesAndTheirExpectedValues.foreach {
+      case (propertyName, expectedValues) =>
+        actualPropertiesAndTheirValues(propertyName).map(_.value) should equal(expectedValues)
+    }
+  }
+
+  "getValuesOfProperties" should "throw an 'NoSuchElementException' if any properties (from which to obtain values from) are not present" in {
+    val namesOfPropertiesToGet = Set("TestProperty2", "TestProperty12", "TestProperty4")
+
+    val thrownException: NoSuchElementException =
+      the[NoSuchElementException] thrownBy customMetadataUtils.getCustomMetadataProperties(namesOfPropertiesToGet)
+
+    thrownException.getMessage should equal("key not found: TestProperty12")
+  }
+
+  "convertPropertiesToFields" should "convert properties to fields for the form, if given correctly formatted properties" in {
+    val propertiesToConvertToFields: Set[CustomMetadata] = allProperties.toSet
+    val fieldValuesByDataType: Set[(FieldValues, String)] = customMetadataUtils.convertPropertiesToFields(propertiesToConvertToFields)
+    val allFieldValues: Map[String, Iterable[FieldValues]] = fieldValuesByDataType.map(_._1).groupBy(_.fieldId)
+
+    propertiesToConvertToFields.foreach{
+      property =>
+        val field = allFieldValues(property.name).head
+
+        field.fieldId should equal(property.name)
+        if(property.dataType == DateTime) field.fieldOptions.length should equal(3) else field.fieldOptions.length should equal(property.values.length)
+        field.fieldHint should equal(property.description.getOrElse(""))
+        field.fieldLabel should equal(property.fullName.get)
+        field.fieldRequired should equal(if(property.propertyGroup.getOrElse("") == "MandatoryMetadata") true else false)
+    }
+  }
+
+  "sortMetadataIntoCorrectPageOrder" should "order the fields in the correct way for the add closure metadata page" in {
```

Comment:
Is this method going to make it into production? If it's just a temporary thing then do you need tests for it?